Tests/FluentTests/QueryHistoryTests.swift

```diff
@@ -0,0 +1,153 @@
+import Fluent
+import Vapor
+import XCTFluent
+import XCTVapor
+
+final class QueryHistoryTests: XCTestCase {
+    func testQueryHistoryDisabled() throws {
+        let app = Application(.testing)
+        defer { app.shutdown() }
+
+        let test = ArrayTestDatabase()
+        app.databases.use(test.configuration, as: .test)
+
+        test.append([
+            TestOutput(["id": 1, "content": "a"]),
+            TestOutput(["id": 2, "content": "b"]),
+        ])
+
+        app.get("foo") { req -> EventLoopFuture<[Post]> in
+            return Post.query(on: req.db).all().map { posts in
+                XCTAssertEqual(req.fluent.history.queries.count, 0)
+                return posts
+            }
+        }
+
+        try app.testable().test(.GET, "foo") { res in
+            XCTAssertEqual(res.status, .ok)
+        }
+    }
+
+    func testQueryHistoryEnabled() throws {
+        let app = Application(.testing)
+        defer { app.shutdown() }
+
+        let test = ArrayTestDatabase()
+        app.databases.use(test.configuration, as: .test)
+
+        test.append([
+            TestOutput(["id": 1, "content": "a"]),
+            TestOutput(["id": 2, "content": "b"]),
+        ])
+
+        app.get("foo") { req -> EventLoopFuture<[Post]> in
+            req.fluent.history.start()
+            return Post.query(on: req.db).all().map { posts in
+                XCTAssertEqual(req.fluent.history.queries.count, 1)
+                return posts
+            }
+        }
+
+        try app.testable().test(.GET, "foo") { res in
+            XCTAssertEqual(res.status, .ok)
+        }
+    }
+
+    func testQueryHistoryEnableAndDisable() throws {
+        let app = Application(.testing)
+        defer { app.shutdown() }
+
+        let test = ArrayTestDatabase()
+        app.databases.use(test.configuration, as: .test)
+
+        test.append([
+            TestOutput(["id": 1, "content": "a"]),
+            TestOutput(["id": 2, "content": "b"]),
+        ])
+
+        app.get("foo") { req -> EventLoopFuture<[Post]> in
+            req.fluent.history.start()
+            return Post.query(on: req.db).all().flatMap { posts -> EventLoopFuture<[Post]> in
+                XCTAssertEqual(req.fluent.history.queries.count, 1)
+                req.fluent.history.stop()
+
+                test.append([
+                    TestOutput(["id": 1, "content": "a"]),
+                    TestOutput(["id": 2, "content": "b"]),
+                ])
+
+                return Post.query(on: req.db).all()
+            }.map { posts in
+                XCTAssertEqual(req.fluent.history.queries.count, 0)
```

Comment:
This should still be `1` since `clear()` was not called, right?

Reply:
There's an unexpected side effect of switching on whether or not enabled is true and returning nil if so - calling `.stop()` clears everything out because it sets `enabled` = false and then the next time .history is called it returns an empty object. I'm not quite sure how to get around this - its almost like we'd need another flag or something.